File: app/src/main/java/libs/com/rogerio/utils/ImgAnimation.java
package libs.com.rogerio.utils;



import java.util.List;

import android.graphics.Canvas;
import android.graphics.drawable.Drawable;

public class ImgAnimation {

	protected List<ObjetosImg> obj;
	protected int max;
	int ind = 0;
	Vector2 position;
	Drawable img;
	ObjetosImg imgO;
	boolean finished;
	public boolean isFinish() {
		return finished;
	}

	public void setFinish(boolean finish) {
		this.finished = finish;
	}

	public ImgAnimation(List<ObjetosImg> objs){
		obj = objs;
		max = obj.size();
		imgO = obj.get(0);
		img = imgO.img;
		finished= false;
	}

	public ImgAnimation() {
		super();
	}

	public ObjetosImg getImg() {		
		return imgO;
	}

	public Vector2 getPosition() {
		return position;
	}

	public void setPosition(Vector2 position) {		
		this.position = position;
		
	}

	public void draw(Canvas canvas) {
		
		img.setBounds((int)position.getX(),(int)position.getY(),
				(int)position.getX()+img.getIntrinsicWidth(),(int)position.getY()+img.getIntrinsicHeight());
		img.draw(canvas);
		
	}

	protected void getObj() {		
		getNewImg();	
		
	}

	public void getNewImg() {
		if((ind+1)==max) finished=true;
		ind = (ind + 1) % max;	
		imgO = obj.get(ind);
		imgO.setPosition(position);
		img = imgO.img;	
	}

	public void moveDown(int mov) {
		position.setY(position.getY()+mov);
		ind = (ind + 1) % max;	
		//getObj();
		
	}

}

File: app/src/main/java/libs/com/rogerio/utils/ImgAnimation.kt
package libs.com.rogerio.utils


import android.graphics.Canvas
import android.graphics.drawable.Drawable

open class ImgAnimation {

    lateinit var obj: List<ObjetosImg>
    var max: Int = 0
    var ind = 0
    var position: Vector2 = Vector2()
    lateinit var img: Drawable
    lateinit var imgO: ObjetosImg
    var isFinish: Boolean = false

    constructor(objs: List<ObjetosImg>) {
        obj = objs
        max = obj.size
        imgO = obj[0]
        img = imgO.img
        isFinish = false
    }

    constructor() : super() {}

    open fun getImg(): ObjetosImg {
        return imgO
    }

    open fun draw(canvas: Canvas) {

        img.setBounds(position.x.toInt(), position.y.toInt(),
                position.x.toInt() + img.intrinsicWidth, position.y.toInt() + img.intrinsicHeight)
        img.draw(canvas)

    }

    protected fun getObj() {
        getNewImg()

    }

    fun getNewImg() {
        if (ind + 1 == max) isFinish = true
        ind = (ind + 1) % max
        imgO = obj[ind]
        imgO.position = position
        img = imgO.img
    }

    fun moveDown(mov: Int) {
        position.y = position.y + mov
        ind = (ind + 1) % max
        //getObj();

    }

}